docs/DeveloperGuide.adoc

```diff
@@ -337,46 +397,232 @@ Use case ends.
++
+Use case resumes at step 1.
+[none]
 
-* ...
-* ...
+'''
+[discrete]
+=== UC06 Generate statistics
+
+*MSS*
+
+1.  User requests to generate statistics.
+2.  MoneyGoWhere generates statistical analysis such as total spending of each tag.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC07 List all tags
+
+*MSS*
+
+1.  User requests to list all the recorded tags.
+2.  MoneyGoWhere lists all the recorded tags.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC08 Import external data from a CSV file
+
+*MSS*
+
+1.  User requests to import data from CSV file by specifying the file path.
+2.  MoneyGoWhere imports the data from the CSV file located at specified file path.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The file path is invalid.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 1.
+[none]
+
+'''
+[discrete]
+=== UC09 Export data to a CSV file
+
+*MSS*
+
+1.  User requests to export data from CSV file by specifying the file path.
+2.  MoneyGoWhere exports the data into the CSV file located at specified file path.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The file path is invalid.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 1.
+[none]
+
+'''
+[discrete]
+=== UC10 Undo a command
+
+*MSS*
+
+1.  User requests to undo a command.
+2.  MoneyGoWhere restores the list of spending to the state before the previous command was executed.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. There are no undo commands executed previously.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC11 Redo a command
+
+*MSS*
+
+1.  User requests to redo a command.
+2.  MoneyGoWhere executes the previous command that was undone.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. There are no undo commands executed previously.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC12 Graphing the data
+
+*MSS*
+
+1.  User requests to display spending in the form of a graph.
+2.  MoneyGoWhere displays spending in graph form.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC13 Clear all spending
+
+*MSS*
+
+1.  User requests to clear all entries in the list of spending.
+2.  MoneyGoWhere deletes all spending entries.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The list of spending is empty.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case ends.
+[none]
+
+[appendix]
+== Non Functional Requirements
+
+.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
+.  Should be able to hold up to 1000 spending without a noticeable sluggishness in performance for typical usage.
+.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
```

Comment:
It is not clear how this NFR would be accomplished. Perhaps an example of a benchmark can be given (e.g. adding an item in 3 seconds when using the keyboard compared to 5 seconds with the GUI).